internal/core/src/query/SearchOnSealed.cpp

```diff
@@ -130,30 +130,35 @@ SearchOnSealed(const Schema& schema,
                 chunk_size);
             bitset_ptr = reinterpret_cast<const uint8_t*>(bitset_data);
         }
-        offset += chunk_size;
         BitsetView bitset_view(bitset_ptr, chunk_size);
 
         if (search_info.group_by_field_id_.has_value()) {
             auto sub_qr = BruteForceSearchIterators(dataset,
                                                     vec_data,
-                                                    row_count,
+                                                    chunk_size,
                                                     search_info,
                                                     bitset_view,
                                                     data_type);
             final_qr.merge(sub_qr);
         } else {
             auto sub_qr = BruteForceSearch(dataset,
                                            vec_data,
-                                           row_count,
+                                           chunk_size,
                                            search_info,
                                            bitset_view,
                                            data_type);
+            for (auto& o : sub_qr.mutable_seg_offsets()) {
```

Comment:
This seems to be super ugly.

can we simply by pass a result the resultset to knowhere and let knowhere fill it inside?

@liliu-z please help on this. also it doesn't make sense to cp the the bitset if it's not aligned